DeviceTests/DeviceTests.Shared/Calendar_Tests.cs

```diff
@@ -33,19 +33,31 @@ public Task Get_Events_List()
             });
         }
 
+        [Theory]
+        [InlineData("ThisIsAFakeId")]
+        [InlineData("-1")]
+        [Trait(Traits.InteractionType, Traits.InteractionTypes.Human)]
+        public Task Get_Events_By_Bad_Calendar_Id(string calendarId)
+        {
+            return Utils.OnMainThread(async () =>
+            {
+                await Assert.ThrowsAsync<ArgumentOutOfRangeException>(() => Calendar.GetEventsAsync(calendarId));
+            });
+        }
+
         [Theory]
         [InlineData("ThisIsAFakeId")]
         [InlineData("-1")]
         [InlineData("")]
         [Trait(Traits.InteractionType, Traits.InteractionTypes.Human)]
-        public Task Get_Event_By_Bad_Id(string calendarId)
+        public Task Get_Event_By_Bad_Id(string eventId)
         {
             return Utils.OnMainThread(async () =>
             {
 #if __IOS__
-                await Assert.ThrowsAsync<NullReferenceException>(() => Calendar.GetEventByIdAsync(calendarId));
+                await Assert.ThrowsAsync<NullReferenceException>(() => Calendar.GetEventByIdAsync(eventId));
```

Comment:
Does this still throw a null ref exception, or can we force this back to an argumentexception like the other platforms?

Reply:
Altered across each platform.

If an invalid argument is passed in, (empty/null) an ArgumentException is thrown, otherwise an ArgumentOutOfRangeException is thrown. The only exception to this case is Android.

Android will also through an ArgumentException when a non-integer id is passed in. Otherwise the behaviour is now shared cross platform. I will have to alter the docs accordingly, but will wait till we're happy with this PR.